### PR TITLE
Fixed Android Issue 863 - LiveQuery is blocked by replicator - Step 1

### LINKED ISSUE #863: Add sample code for FDB store

### PR DESCRIPTION
**Pull request by hideki**
Canceling HttpClient request instead of waiting  HTTP request finish.

### CODE REVIEW

File: src/main/java/com/couchbase/lite/Database.java [hunk 1]
```diff
@@ -1280,18 +1280,20 @@ public boolean close() {
...
                 replicator.stop();
+                stopping = true;
             }
 
             // maximum wait time per replicator is 60 sec.
             // total maximum wait time for all replicators is between 60sec and 119 sec.
             long timeout = Replication.DEFAULT_MAX_TIMEOUT_FOR_SHUTDOWN * 1000;
             long startTime = System.currentTimeMillis();
-            while (activeReplicators.size() > 0 &&
+            while (activeReplicators.size() > 0 && stopping &&
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/Database.java]: Just a question. After this fix, the recommended MAX timeout is still 60 or can it be reduced to a very small amount?

File: src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java [hunk 2]
```diff
@@ -1781,16 +1805,18 @@ protected static void waitBatcherCompleted(Batcher<RevisionInternal> b) {
...
+            while (!pendingFutures.isEmpty()) {
+                Future future = pendingFutures.take();
                 try {
                     future.get();
                 } catch (InterruptedException e) {
                     Log.e(Log.TAG_SYNC, "InterruptedException in Future.get()", e);
                 } catch (ExecutionException e) {
                     Log.e(Log.TAG_SYNC, "ExecutionException in Future.get()", e);
+                } finally {
+                    runnables.remove(future);
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java]: I couldn't follow all the logics from reading the code here. One quick question. Will there be a chance that there will be some runnables not getting removed from the runnables map as the tasks finish before waitPendingFuturesCompleted() is called?

File: src/main/java/com/couchbase/lite/util/CancellableRunnable.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.couchbase.lite.util;
+
+/**
+ * Created by hideki on 5/5/16.
+ */
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/util/CancellableRunnable.java]: Should we have a complete license header?

File: src/main/java/com/couchbase/lite/replicator/PullerInternal.java [hunk 4]
```diff
@@ -178,6 +178,11 @@ protected void startChangeTracker() {
     protected void processInbox(RevisionList inbox) {
         Log.d(TAG, "processInbox called");
 
+        if(db == null||!db.isOpen()) {
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/replicator/PullerInternal.java]: Formatting.

File: src/main/java/com/couchbase/lite/Database.java (hunk 1 above)
Comment (hideki) [inline src/main/java/com/couchbase/lite/Database.java]: Not necessary to wait 60 sec. I am considering to use default 20 sec.

File: src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java (hunk 2 above)
Comment (hideki) [inline src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java]: `Runnable` should be removed when task completed.  It should not be removed before `waitPendingFuturesCompleted()`.

Unprocessed `Runnable`s will be canceled and removed from runnables in `stop()` or `close()` methods

**Pull request merged**